test/compression.test

```diff
@@ -106,11 +106,15 @@ for level in $levels; do
   wait_script foo hosts/bar-up
   wait_script bar hosts/foo-up
 
-  ip netns exec foo \
-    socat -u TCP4-LISTEN:$recv_port_foo,reuseaddr OPEN:"$tmp_file",creat &
-
-  ip netns exec bar \
-    socat -u OPEN:"$ref_file" TCP4:$ip_foo:$recv_port_foo,retry=30 &
+  (
+    ip netns exec foo \
+      socat -u TCP4-LISTEN:$recv_port_foo,reuseaddr OPEN:"$tmp_file",creat || kill $$
+  ) &
+
+  (
+    ip netns exec bar \
+      socat -u OPEN:"$ref_file" TCP4:$ip_foo:$recv_port_foo,retry=30 || kill $$
+  ) &
 
   wait
```

Comment:
This will still hang indefinitely if any of the socat commands don't terminate. Maybe it's better to just do something like:

```
cat > $tmp_file.sh << EOF
ip netns exec foo socat -u TCP4-LISTEN:$recv_port_foo,reuseaddr OPEN:"$tmp_file",creat &
ip netns exec bar socat -u OPEN:"$ref_file" TCP4:$ip_foo:$recv_port_foo,retry=30 &
wait
EOF

try_limit_time 60 sh -c $tmp_file.sh
```

Reply:
Yes, this looks much better. Thank you.

It still could use the faster error detection (`|| kill $$`), but it's much uglier, so I removed it.

This all could have been written simpler (and it was), but of course there's old korn shell which handles background jobs unlike every other shell I've tested.